### ISSUE TITLE
Weird behaviors from using OpenRouter's models

### ISSUE THREAD

**Issue by Koihin88** [yetone/avante.nvim#186]
Thank you for this amazing plugin!

I've encountered some strange behaviors while using OpenRouter's AI models. Here are the details:

This is my config file:
```lua
return {
  {
    "yetone/avante.nvim",
    event = "VeryLazy",
    build = "make",
    opts = {
      -- add any opts hereby
      provider = "openai",
      openai = {
        endpoint = "https://openrouter.ai/api",
        model = "openai/gpt-4o-2024-08-06",
        temperature = 0,
        max_tokens = 4096,
      },
    },
    dependencies = {
      "nvim-tree/nvim-web-devicons",
      "stevearc/dressing.nvim",
      "nvim-lua/plenary.nvim",
      "MunifTanjim/nui.nvim",
      --- The below is optional, make sure to setup it properly if you have lazy=true
      {
        "MeanderingProgrammer/render-markdown.nvim",
        opts = {
          file_types = { "markdown", "Avante" },
        },
        ft = { "markdown", "Avante" },
      },
    },
  },
}

```
---

1. Issue with `openai/gpt-4o-2024-08-06` and `meta-llama/llama-3.1-405b`:
   - The sidebar toggles off automatically after generating a few lines.
   - When toggling the sidebar back on, a duplicate of the conversation appears.
   - This occurs despite the model API being called only once.

```json
[
   {
      "provider":"openai",
      "timestamp":"2024-08-24 13:39:33",
      "model":"openai/gpt-4o-2024-08-06",
      "response":"Replace lines: 23-37\n```python\n            # ImageMagick command to convert APNG to GIF\n            cmd = [\n                \"convert\",\n                input_path,\n                \"-coalesce\",\n                \"-layers\",\n                \"optimize\",\n                output_path,\n            ]\n\n            # Run the ImageMagick command\n            subprocess.run(\n                cmd, check=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL\n            )\n```",
      "request":"use imagemagick instead"
   },
   {
      "provider":"openai",
      "timestamp":"2024-08-24 13:39:33",
      "model":"openai/gpt-4o-2024-08-06",
      "response":"Replace lines: 23-37\n```python\n            # ImageMagick command to convert APNG to GIF\n            cmd = [\n                \"convert\",\n                input_path,\n                \"-coalesce\",\n                \"-layers\",\n                \"optimize\",\n                output_path,\n            ]\n\n            # Run the ImageMagick command\n            subprocess.run(\n                cmd, check=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL\n            )\n```",
      "request":"use imagemagick instead"
   }
]
```
<img width="452" alt="Screenshot 2024-08-24 at 14 01 10" src="https://github.com/user-attachments/assets/b81a0dd6-81d4-42ba-a516-b3fba5d6f356">

2. Issue with `anthropic/claude-3.5-sonnet`:
   - Unable to use `<Tab>` to continue the conversation after generation is complete.
   - Toggle the sidebar off and on only to find out the chat message to disappear.
   - The `.avante_chat_history` folder is not created.
<img width="453" alt="Screenshot 2024-08-24 at 14 16 24" src="https://github.com/user-attachments/assets/b99981ba-5e7e-4407-9da2-a97b52e20042">

**Comment by yetone** [comment]
<img width="527" alt="image" src="https://github.com/user-attachments/assets/4847f27b-bb08-4a90-b9f5-76b2721912d4">

When this line appears, it indicates that the content generation is complete. If this line does not appear, it means the content is still being generated. At this time, the input will disappear and no history record will be created.

**Comment by yetone** [comment]
> The sidebar toggles off automatically after generating a few lines.

Pressing the q or <Esc> key will close the sidebar. Did you press these keys?

**Comment by Koihin88** [comment]
> When this line appears, it indicates that the content generation is complete. If this line does not appear, it means the content is still being generated.

I suspect the models from OpenRouter do not work very well with the system prompt in this plugin? Because they don't have `🎉🎉🎉 **Generation complete!** Please review the code suggestions above.` at the end of the responses. And when I use `anthropic/claude-3.5-sonnet`, it finishes generating a full response but then just freezes up. (This freezing problem is only happening with the `anthropic/claude-3.5-sonnet` though.)


> Pressing the q or key will close the sidebar. Did you press these keys?

No, I did not touch anything.

**Comment by yetone** [comment]
‌‌‌This issue has been fixed in the latest version, you can give it a try.

**Comment by Koihin88** [comment]
🧎‍♂️You are shipping so fast, what a legend! Everything is on point except the `anthropic/claude-3.5-sonnet` model. It does not follow your prompt very strictly. And despite finishing generating its response, the `.avante_chat_history` was not created either. I think I will just stick with other models for now. Thank you!
